oauth2/generic_test.go

```diff
@@ -0,0 +1,107 @@
+package oauth2_test
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	clog "github.com/influxdata/chronograf/log"
+	"github.com/influxdata/chronograf/oauth2"
+)
+
+func TestGenericPrincipalID(t *testing.T) {
+	t.Parallel()
+
+	response := struct {
+		Email string `json:"email"`
+	}{
+		"martymcfly@pinheads.rok",
+	}
+	mockAPI := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
```

Comment:
I think that's true in practice, however, from a testing point of view this code isn't testing if the provider server is in spec but if it can communicate with that server.